### PR TITLE
Implement setProrotypeOf function

### PR DESCRIPTION
**Pull request by VenelinBakalov**
### Description
Implement Object.setPrototypeOf() method

### Checklist

- [ ] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation, relevant usage information (if applicable)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have tested against live environment, if applicable
- [ ] I have synced any structure and/or content vRA-NG improvements with vra-ng and ts-vra-ng archetypes (if applicable)
- [ ] I have my changes rebased and squashed to the minimal number of relevant commits. **Notice: don't squash all commits**
- [x] I have added a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### Related issues and PRs

#264

### CODE REVIEW

**Comment by VenelinBakalov** [comment]
Tested with the following code samples:
```
 var Animal = function () { }
        Animal.prototype.speak = function () {
            System.log(this.name + ' makes a noise.');
        }

        var Dog = function (name) {
            this.name = name
        }

        // set up prototype chain
        Object.setPrototypeOf(Dog.prototype, Animal.prototype);

        // Legacy approach
        // Dog.prototype.__proto__=Animal.prototype

        var d = new Dog('Mitzie');
        d.speak(); // Mitzie makes a noise.


        class NsxtError extends Error {
            public readonly cause: string
            constructor(message: string, service: string) {
                super(`Received error ${message} from NSX-T Service ${service}.`);
                Object.setPrototypeOf(this, new.target.prototype)
            }
        }

        const testError = new NsxtError("Object not found", "SecurityPolicyService");
        System.log(`Constructor name: ${testError.constructor.name}`);
        System.error("=========")
        class NsxtError2 extends Error {
            public readonly cause: string
            constructor(message: string, service: string) {
                super(`Received error ${message} from NSX-T Service ${service}.`);
            }
        }

        const testError2 = new NsxtError2("Object not found", "SecurityPolicyService");
        System.log(`Constructor name: ${testError2.constructor.name}`);
```

File: typescript/vrotsc/src/compiler/transformer/codeTransformers/shims.ts [hunk 1]
```diff
@@ -62,150 +76,153 @@ interface ShimReference {
...
+                case "Array.fill":
+                    return shimInstanceCall("arrayFill", node);
+                case "ArrayConstructor.from":
+                    return shimStaticCall("arrayFrom", node);
+                case "ArrayConstructor.of":
+                    return shimStaticCall("arrayOf", node);
+                case "ObjectConstructor.assign":
+                    return shimStaticCall("objectAssign", node);
+                case "ObjectConstructor.setPrototypeOf":
+                    console.error("TEST!!!");
```
Comment (Michaelpalacce) [inline typescript/vrotsc/src/compiler/transformer/codeTransformers/shims.ts]: Remove

**Review by Michaelpalacce — DISMISSED** [review]
Can you also make sure to use tabs?

File: typescript/vrotsc/src/compiler/transformer/codeTransformers/shims.ts (hunk 1 above)
Comment (VenelinBakalov) [reply typescript/vrotsc/src/compiler/transformer/codeTransformers/shims.ts]: Yeap I saw that with a bit of delay hahah

**Review by Michaelpalacce — APPROVED** [review]

**Review by akantchev — APPROVED** [review]

**Pull request merged**